server/map_tree_updater.go

```diff
@@ -20,26 +20,35 @@ import (
 	"sync"
 
 	"github.com/google/trillian"
+	"github.com/google/trillian/extension"
 	"github.com/google/trillian/merkle"
 	"github.com/google/trillian/merkle/hashers"
 	"github.com/google/trillian/storage"
 	"github.com/google/trillian/storage/tree"
 )
 
-// updateTree updates the sparse Merkle tree at the specified revision based on the passed-in
-// leaf changes, and writes it to the storage. Returns the new signed map root, which is also
-// submitted to storage.
-func (t *TrillianMapServer) updateTree(ctx context.Context, tree *trillian.Tree, hasher hashers.MapHasher, tx storage.MapTreeTX, hkv []merkle.HashKeyValue, rev int64) ([]byte, error) {
+// mapTreeUpdater updates the sparse Merkle tree of the map in one or multiple
+// map tree transactions.
+type mapTreeUpdater struct {
+	tree   *trillian.Tree
+	hasher hashers.MapHasher
+	opts   TrillianMapServerOptions
+	reg    extension.Registry
```

Comment:
Recommend swapping this for MapStorage. I'm a big fan of giving dependencies the smallest possible surface area so we can reason about what's going on. It also makes it clearer what the intent is.

Reply:
Done.